packages/react-native/src/posthog-rn.ts

```diff
@@ -742,9 +753,14 @@ export class PostHog extends PostHogCore {
    * @public
    *
    * @param properties The group properties to set for flag evaluation
+   * @param reloadFeatureFlags Whether to reload feature flags after setting the properties. Defaults to true.
    */
-  setGroupPropertiesForFlags(properties: Record<string, Record<string, string>>): void {
+  setGroupPropertiesForFlags(properties: Record<string, Record<string, string>>, reloadFeatureFlags = true): void {
     super.setGroupPropertiesForFlags(properties)
+
+    if (reloadFeatureFlags) {
+      this.reloadFeatureFlags()
+    }
```

Comment:
Regarding existing cases where users may manually `reloadFeatureFlags` themselves after calling this method:

1. `reloadFeatureFlags` cannot be awaited
https://github.com/PostHog/posthog-js/blob/d331ab6ec2b8e1ce4e8f04623bbabcb0f7413154/packages/core/src/posthog-core.ts#L817-L818

2. There is a guard in place to prevent more than one concurrent request
https://github.com/PostHog/posthog-js/blob/d331ab6ec2b8e1ce4e8f04623bbabcb0f7413154/packages/core/src/posthog-core.ts#L453-L460

Given these two things, it should mitigate against a double reload.